### PR TITLE
py-typed-ast: add version 1.5.0, which builds on macOS with clang

### PR DESCRIPTION
**Pull request by dev-zero**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-typed-ast/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyTypedAst(PythonPackage):
     homepage = "https://github.com/python/typed_ast"
     pypi = "typed-ast/typed_ast-1.4.0.tar.gz"
 
+    version('1.5.0', sha256='ff4ad88271aa7a55f19b6a161ed44e088c393846d954729549e3cde8257747bb')
     version('1.4.3', sha256='fb1bbeac803adea29cedd70781399c99138358c26d05fcbd23c13016b7f5ec65')
     version('1.4.2', sha256='9fc0b3cb5d1720e7141d103cf4819aea239f7d136acf9ee4a69b047b7986175a')
     version('1.4.1', sha256='8c8aaad94455178e3187ab22c8b01a3837f8ee50e09cf31f1ba129eb293ec30b')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-typed-ast/package.py]: Can you add:
```python
depends_on('python@3.6:', when='@1.5:', type=('build', 'link', 'run'))
```
and also specify the dep type for the other python dep that's missing it?

File: var/spack/repos/builtin/packages/py-mypy/package.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ class PyMypy(PythonPackage):
 
     depends_on("python@3.5:", type=("build", "run"))
     depends_on('py-setuptools', type=('build', 'run'))
-    depends_on('py-typed-ast@1.4.0:1.4', type=('build', 'run'))
+    depends_on('py-typed-ast@1.4.0:2', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-mypy/package.py]: According to the setup.py:
```python
      install_requires=["typed_ast >= 1.4.0, < 1.5.0; python_version<'3.8'",             
````
this is right, although it needs a small tweak:
```suggestion
    depends_on('py-typed-ast@1.4.0:1.4', when='@0.900: ^python@:3.7', type=('build', 'run'))
    depends_on('py-typed-ast@1.4.0:1.4', when='@:0.899', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-mypy/package.py (hunk 2 above)
Comment (dev-zero) [reply var/spack/repos/builtin/packages/py-mypy/package.py]: was looking only at the latest `setup.py`, and ignored the Python conditional, my bad

File: var/spack/repos/builtin/packages/py-typed-ast/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-typed-ast/package.py]: @dev-zero

Comment (dev-zero) [reply var/spack/repos/builtin/packages/py-typed-ast/package.py]: done, thanks for the reminder

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
Can you resolve the merge conflicts on this? We should get this merged.

**Comment by adamjstewart** [comment]
@spackbot fix style

**Comment by alalazo** [comment]
We have v1.5.4 on the current `develop` so closing

**Pull request closed without merging**